pkgs/applications/networking/p2p/lokinet/default.nix

```diff
@@ -0,0 +1,69 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, cmake
+, curl
+, libevent
+, libsodium
+, libuv
+, pkgconfig
+, sqlite
+, unbound
+, writeShellScriptBin
+, zeromq
+}:
+
+stdenv.mkDerivation rec {
+  pname = "lokinet";
+  version = "0.9.9";
+  githash = "440b547d2c3e7384ec82ec21f0b1635e42bcda64";
+
+  src = fetchFromGitHub {
+    owner = "oxen-io";
+    repo = "lokinet";
+    rev = "v${version}";
+    sha256 = "AaGsRg9S9Cng9emI/mN09QSOIRbE+x3916clWAwLnRs=";
+    fetchSubmodules = true;
+  };
+
+  fakegit = writeShellScriptBin "git" "echo ${githash}";
+
```

Comment:
```suggestion
```
Patching that would be easier. Can we easily do that?

Reply:
After a little digging, it seems I can specify the git version by adding a cmake flag :
```nix
cmakeFlags = [
  "-DGIT_VERSION=hash"
];
```
Now, how would I go about specifying the version automatically ? I tried this :

Reply:
That following is good enough for us and doesn't involve any extra work when updating.

```nix
cmakeFlags = [
  "-DGIT_VERSION=v${version}"
];
```